tests/phpunit/tests/post/wpGetAutoSave.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @group post
+ */
+class Tests_Post_GetPostAutosave extends WP_UnitTestCase {
+
+	/**
+	 * Admin user ID.
+	 *
+	 * @var int
+	 */
+	protected static $admin_id;
+
+	/**
+	 * Editor user ID.
+	 *
+	 * @var int
+	 */
+	protected static $editor_id;
+
+	/**
+	 * Post ID.
+	 *
+	 * @var int
+	 */
+	protected static $post_id;
+
+	/**
+	 * Ticket #62658
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/62658
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$admin_id  = $factory->user->create( array( 'role' => 'administrator' ) );
+		self::$editor_id = $factory->user->create( array( 'role' => 'editor' ) );
+
+		wp_set_current_user( self::$admin_id );
+		self::$post_id = $factory->post->create( array( 'post_status' => 'publish' ) );
+	}
+
+	/**
+	 * Test when no autosave exists for a post.
+	 */
+	public function test_no_autosave_exists() {
+		$autosave = wp_get_post_autosave( self::$post_id );
+		$this->assertFalse( $autosave, 'Expected no autosave.' );
+	}
+
+	/**
+	 * Test when an autosave exists for a post.
+	 */
+	public function test_autosave_exists() {
+		$autosave_id = wp_insert_post(
+			array(
+				'post_type'    => 'revision',
+				'post_status'  => 'inherit',
+				'post_parent'  => self::$post_id,
+				'post_author'  => self::$admin_id,
+				'post_content' => 'Autosaved content',
+				'post_name'    => self::$post_id . '-autosave-v1',
+			)
+		);
+
+		$autosave = wp_get_post_autosave( self::$post_id );
+
+		$this->assertInstanceOf( 'WP_Post', $autosave );
+		$this->assertSame( $autosave_id, $autosave->ID, 'Autosave ID does not match.' );
+		$this->assertSame( self::$post_id, (int) $autosave->post_parent, 'Post parent ID does not match.' );
+	}
+
+	/**
+	 * Test when an autosave exists for a specific user.
+	 */
```

Comment:
```suggestion
	/**
	 * Test when an autosave exists for a specific user.
	 *
	 * @ticket 62658
	 */
```

Reply:
@mukeshpanchal27 Thanks for the feedback, I have implemented the suggestions.